doc/custom_data.rst

```diff
@@ -0,0 +1,118 @@
++===============================+========================================+
+| Gas network                   | ``resources/custom_data/pipelines.csv``|
++-------------------------------+----------------------------------------+
+| Export ports                  | ``data/custom/export_ports.csv``       |
++-------------------------------+----------------------------------------+
+| Airports                      | ``data/custom/airports.csv``           |
++-------------------------------+----------------------------------------+
+| Powerplants                   | ``data/custom_powerplants.csv``        |
++-------------------------------+----------------------------------------+
+| Demand/Renewables Time Series | ``.nc`` files compatible with          |
+|                               | GEGIS/atlite modules                   |
++-------------------------------+----------------------------------------+
+
+.. note::
+
+   Custom datasets should follow the filename conventions specified by PyPSA-Earth to ensure proper integration.
+
+Reference Data Sources
+---------------------
+
+For guidance on sourcing data, refer to the following table:
+
++------+----------------------------------------+--------+--------+---------+-----+
+| Name | Link                                   | Sector | Global | Country | API |
++======+========================================+========+========+=========+=====+
+| IEA  | https://www.iea.org/countries/         | All    | ✔      |         | ?   |
++------+----------------------------------------+--------+--------+---------+-----+
+| WRI  | https://www.wri.org/                   | All    | ✔      |         | ?   |
++------+----------------------------------------+--------+--------+---------+-----+
+| OECD | https://stats.oecd.org/                | All    | ✔      |         | ?   |
++------+----------------------------------------+--------+--------+---------+-----+
+
+.. note::
+
+   This table is continuously updated to include new global and country-level datasets.
+
+Best Practices
+--------------
+
+- Keep custom datasets in the recommended directories to avoid conflicts
+- Maintain the same format and prescribed filenames as the default CSV/NetCDF files for seamless integration
+- Document any assumptions or modifications made in custom data for future reproducibility
+
+
+Additional Notes
+----------------
+
+- If using **GADM clustering**, ensure at least one bus per administrative region. Missing buses can be added using a custom CSV created with centroids matching the substation GeoJSON format.
+- Private datasets do not need to be shared publicly.
+- Users are encouraged to contribute improvements back to the repository following contribution guidelines.
+
+Usage Instructions
+------------------
+
+1. Enable the desired options in ``config.default.yaml``.
+2. Place required custom CSV/NetCDF files in the specified directories.
+3. Integrate demand/renewable time series following documentation instructions.
```

Comment:
1. You could also add the reference here also; [demand section](https://pypsa-earth.readthedocs.io/en/latest/customization_basic1.html#use-custom-demand-data)
2. I also think it is repetition to refer to the documentation. I think `Integrate demand/renewable time series following the instructions.` should suffice. 

Feel free to adjust it as you see fit.